Modules/_pickle.c

```diff
@@ -1639,9 +1639,9 @@ _Unpickler_SetInputStream(UnpicklerObject *self, PyObject *file)
     if (_PyObject_LookupAttr(file, &_Py_ID(readinto), &self->readinto) < 0) {
         return -1;
     }
-    (void)_PyObject_LookupAttr(file, &_Py_ID(read), &self->read);
-    (void)_PyObject_LookupAttr(file, &_Py_ID(readline), &self->readline);
-    if (!self->readline || !self->read) {
+    if (_PyObject_LookupAttrId(file, &PyId_read, &self->read) <= 0 ||
```

Comment:
Seems that you forgot to define a `PyId_read` and `PyId_readline`.
However, have you try to a built python with this patch manually?

Reply:
####  I have fixed this issue in the latest commit ✅
####  Please have a look , Thanks

Reply:
Seems that tests keep falling.
You sure that you made necessary changes?

Reply:
## Below changes are made by me : 
![image](https://user-images.githubusercontent.com/40736800/229489041-63a1a164-1636-4fa1-b79a-ead1e9ae5e4f.png)


## All other changes are imported as it is from https://github.com/python/cpython/pull/16606

## Therefore, I am not sure where exactly the code is failing .

Reply:
Check the CI/CD.